kurtosis/kurtosis/servicesavalanche/avalanchegonode/avalanchego_container_config_factory.go

```diff
@@ -0,0 +1,138 @@
+package avalanchegonode
+
+import (
+	"bytes"
+	"fmt"
+	"github.com/ava-labs/avalanchego-kurtosis/kurtosis/avalanche/libs/builder/networkbuilder"
+	"github.com/ava-labs/avalanchego-kurtosis/kurtosis/avalanche/libs/constants"
+	"github.com/kurtosis-tech/kurtosis-libs/golang/lib/services"
+	"github.com/palantir/stacktrace"
+	"github.com/sirupsen/logrus"
+	"os"
+	"strings"
+)
+
+const (
+	testVolumeMountpoint = "/test-volume"
+	configFileID         = "cChainConfig"
+	configFileContents   = `{"coreth-config":{"snowman-api-enabled": false,"coreth-admin-api-enabled": false,"net-api-enabled": true,"rpc-gas-cap": 2500000000,"rpc-tx-fee-cap": 100,"eth-api-enabled": true,"personal-api-enabled": true,"tx-pool-api-enabled": true,"debug-api-enabled": false,"web3-api-enabled": true,"local-txs-enabled": true}}`
+)
+
+type AvalancheGoContainerConfigFactory struct {
+	nodeConfig             *networkbuilder.Node
+	definedNetwork   *networkbuilder.Network
+	createdNodes     map[services.ServiceID]*NodeAPIService
+	bootstrapNodes   string
+	bootstrapNodeIPs string
+}
+
+func NewAvalancheGoContainerConfigFactory(definedNetwork *networkbuilder.Network, nodeConfig *networkbuilder.Node, nodes map[services.ServiceID]*NodeAPIService) *AvalancheGoContainerConfigFactory {
+	return &AvalancheGoContainerConfigFactory{definedNetwork: definedNetwork, nodeConfig: nodeConfig, createdNodes: nodes}
+}
+
+func (factory AvalancheGoContainerConfigFactory) GetCreationConfig(containerIpAddr string) (*services.ContainerCreationConfig, error) {
```

Comment:
Can this `services.ContainerCreationConfig` be an interface in the future ?

Reply:
What's the worry with this? The `AvalancheGoContainerConfigFactory` is explicitly an implementation of a Kurtosis interface, `ContainerConfigFactory`, which is the same as how the old `NodeInitializer` was a direct implementation of Kurtosis interface `DockerContainerInitializer` - there's not really a way to un-tie this from Kurtosis since the goal of this class is explicitly to implement a Kurtosis interface